Building docker images using local registry

### Description

There is a pull rate limit recently introduced by dockerhub, therefore any automated jenkins builds start failing with
```
17:55:16  Step 3/21 : FROM golang:${GO_VERSION}-alpine${ALPINE_VERSION} as go-builder
17:55:16  toomanyrequests: You have reached your pull rate limit. You may increase the limit by authenticating and upgrading: https://www.docker.com/increase-rate-limit
17:55:16  Makefile:34: recipe for target 'build-image' failed
17:55:16  make: *** [build-image] Error 1
```
### Importance

Should-have

### Location

Dockerfile and Makefile

### Suggestions for an improvement

Builds should pass

--- a/Dockerfile
+++ b/Dockerfile
@@ -7,10 +7,11 @@
 #
 #     http://www.apache.org/licenses/LICENSE-2.0
 #
+ARG DOCKER_REGISTRY
 ARG GO_VERSION=1.13.8
 ARG ALPINE_VERSION=3.11
 
-FROM golang:${GO_VERSION}-alpine${ALPINE_VERSION} as go-builder
+FROM ${DOCKER_REGISTRY:+$DOCKER_REGISTRY/}golang:${GO_VERSION}-alpine${ALPINE_VERSION} as go-builder
 
 ARG PROJECT_NAME=bookkeeper-operator
 ARG REPO_PATH=github.com/pravega/$PROJECT_NAME
@@ -33,7 +34,7 @@ RUN GOOS=linux GOARCH=amd64 CGO_ENABLED=0 go build -o /src/${PROJECT_NAME} \
     /src/cmd/manager
 
 # =============================================================================
-FROM alpine:${ALPINE_VERSION} AS final
+FROM ${DOCKER_REGISTRY:+$DOCKER_REGISTRY/}alpine:${ALPINE_VERSION} AS final
 
 RUN apk add --update \
     sudo \

--- a/Makefile
+++ b/Makefile
@@ -32,7 +32,7 @@ build-go:
 
 build-image:
 	echo "$(REPO)"
-	docker build --no-cache --build-arg VERSION=$(VERSION) --build-arg GIT_SHA=$(GIT_SHA) -t $(REPO):$(VERSION) .
+	docker build --no-cache --build-arg VERSION=$(VERSION) --build-arg DOCKER_REGISTRY=$(DOCKER_REGISTRY) --build-arg GIT_SHA=$(GIT_SHA) -t $(REPO):$(VERSION) .
 	docker tag $(REPO):$(VERSION) $(REPO):latest
 
 test: test-unit test-e2e

